Source/Caches/HACacheKeyStates.swift

```diff
@@ -0,0 +1,65 @@
+import Foundation
+
+/// Key for the cache
+internal struct HACacheKeyStates: HACacheKey {
+    static func create(connection: HAConnection) -> HACache<HACachedStates> {
+        .init(
+            connection: connection,
+            subscribe:
+            .init(
+                subscription: .subscribeEntities(),
+                transform: { info in
+                    .replace(processUpdates(info: info))
+                }
+            )
+        )
+    }
+
+    /// Process updates from the compressed state to HAEntity
+    /// - Parameter info: The compressed state update and the current cached states
+    /// - Returns: HAEntity cached states
+    /// Logic from: https://github.com/home-assistant/home-assistant-js-websocket/blob/master/lib/entities.ts
+    static func processUpdates(info: HACacheTransformInfo<HACompressedStatesUpdates, HACachedStates?>)
+        -> HACachedStates {
+        var states = info.current ?? .init(entities: [])
```

Comment:
What happens if we're starting a new subscription and we get back the 'initial' value set. Do we handle the case where: old value contains entity X, new subscription's first result doesn't include X - does X continue to show? I think it does from how this code path works.

Reply:
This may be a reason to 'populate' (but always set the value to nil) the cache for this particular state, which might require changes to the other PR, too. Or perhaps a flag to let it know, in this case, that it's coming from stale data. Hmm.

Reply:
If X is not included in the second update it's going to be removed by the "processUpdate" right? during the "subtractions" check on line 41

Reply:
If the old value is from an hour ago, the initial subscription isn't going to include it as a deletion because it doesn't know you want to delete it.